src/Microsoft.AspNet.Hosting.Interfaces/IHostingEnvironment.cs

```diff
@@ -7,10 +7,13 @@ namespace Microsoft.AspNet.Hosting
 {
     public interface IHostingEnvironment
     {
+        // This must be settable!
         string EnvironmentName { get; set;  }
```

Comment:
Can you change this in the application so that you start off as dev then end up as prod? Is that what we want?